### ISSUE TITLE
Found torch.nn.functional.fractional_max_pool2d has inconsistency bug between GPU and CPU testing

### ISSUE THREAD

**Issue by beanduan22** [pytorch/pytorch#113844]
### 🐛 Describe the bug

import torch
import torchvision
import torchvision.transforms as transforms
import torch.nn as nn
import os
import torch.nn.functional as F
gpu_device = torch.device("cuda")
cpu_device = torch.device("cpu")
class PreprocessAndCalculateModel(nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, x):
        # = x.view(x.size(0), -1)
        output = F.fractional_max_pool2d(x, kernel_size=3, output_ratio=(0.5, 0.5))
        return output

transform = transforms.Compose([transforms.ToTensor()])
cifar_dataset = torchvision.datasets.CIFAR10(root='./data', train=True, transform=transform, download=True)
data_loader = torch.utils.data.DataLoader(cifar_dataset, batch_size=64, shuffle=True)

    # 从数据集中获取一个批次的图像作为输入
real_inputs, real_targets = next(iter(data_loader))
model = PreprocessAndCalculateModel()
x = real_inputs
output_gpu = model.to(gpu_device)(x.cuda())
output_cpu = model.to(cpu_device)(x.cpu())
print(torch.allclose(output_gpu.cpu(), output_cpu, atol=1e-3))

### Versions

2.0.0

**Comment by janeyx99** [comment]
Closing this issue as numerical accuracy differences are expected across devices, especially when mathematical operations compound. See https://pytorch.org/docs/stable/notes/numerical_accuracy.html. Usually, the difference is negligible and should not affect convergence. If that is not the case, please do not hesitate to follow up and reopen the issue!
